kubectl-plugin/pkg/cmd/session/session.go

```diff
@@ -0,0 +1,126 @@
+package session
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubectl/pkg/cmd/portforward"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	DASHBOARD_PORT = 8265
+	CLIENT_PORT    = 10001
+)
+
+type SessionOptions struct {
+	ioStreams    *genericiooptions.IOStreams
+	configFlags  *genericclioptions.ConfigFlags
+	ResourceName string
+	Namespace    string
+}
+
+func NewSessionOptions(streams genericiooptions.IOStreams) *SessionOptions {
+	return &SessionOptions{
+		ioStreams:   &streams,
+		configFlags: genericclioptions.NewConfigFlags(true),
+	}
+}
+
+func NewSessionCommand(streams genericiooptions.IOStreams) *cobra.Command {
+	options := NewSessionOptions(streams)
+	factory := cmdutil.NewFactory(options.configFlags)
+
+	cmd := &cobra.Command{
+		Use:   "session NAME",
+		Short: "Forward local ports to the Ray resources. Currently only supports RayCluster.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := options.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := options.Validate(); err != nil {
+				return err
+			}
+			return options.Run(cmd.Context(), factory)
+		},
+	}
+	options.configFlags.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (options *SessionOptions) Complete(cmd *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return cmdutil.UsageErrorf(cmd, "%s", cmd.Use)
+	}
+	options.ResourceName = args[0]
+
+	if *options.configFlags.Namespace == "" {
+		options.Namespace = "default"
+	} else {
+		options.Namespace = *options.configFlags.Namespace
+	}
+
+	return nil
+}
+
+func (options *SessionOptions) Validate() error {
+	// Overrides and binds the kube config then retrieves the merged result
+	config, err := options.configFlags.ToRawKubeConfigLoader().RawConfig()
+	if err != nil {
+		return fmt.Errorf("Error retrieving raw config: %w", err)
+	}
+	if len(config.CurrentContext) == 0 {
+		return fmt.Errorf("no context is currently set, use %q to select a new one", "kubectl config use-context <context>")
+	}
+	return nil
+}
+
+func (options *SessionOptions) Run(ctx context.Context, factory cmdutil.Factory) error {
+	kubeClientSet, err := factory.KubernetesClientSet()
+	if err != nil {
+		return fmt.Errorf("failed to initialize clientset: %w", err)
+	}
+
+	svcName, err := findServiceName(ctx, kubeClientSet, options.Namespace, options.ResourceName)
+	if err != nil {
+		return err
+	}
+
+	portForwardCmd := portforward.NewCmdPortForward(factory, *options.ioStreams)
```

Comment:
@andrewsykim @MortalHappiness Want to make sure I'm on the same page, since I know there were some discussion on the port forwarding. Is the plan to currently use kubectl port forward since it's more straightfoward and change to client-go in the future for more configurability?

Reply:
Yes, now in https://github.com/ray-project/kuberay/pull/2298 I use the `kubectl port-forward` directly by calling `ExecuteContext` (from `cobra` package).

https://pkg.go.dev/github.com/spf13/cobra#Command.ExecuteContext